src/test_main.py

```diff
@@ -59,35 +60,231 @@ def test_generate_state_space_for_N_eq_1_and_k_eq_3():
+        )
+        == 0
+    )
+    source = np.array((0, 0, 0))
+    target = np.array((0, 0, 0))
+    assert (
+        main.compute_transition_probability(
+            source=source, target=target, fitness_function=trivial_fitness_function
+        )
+        is None
+    )
+
+
+def test_compute_transition_probability_for_specific_fitness_function():
+    """"""
+
+    def fitness_function(state):
+        return np.array([np.count_nonzero(state == _) for _ in state])
+
+    source = np.array((0, 1, 0))
+    target = np.array((1, 1, 0))
+    assert (
+        main.compute_transition_probability(
+            source=source, target=target, fitness_function=fitness_function
+        )
+        == 1 / 15
+    )
+    source = np.array((0, 1, 1))
+    target = np.array((0, 0, 0))
+    assert (
+        main.compute_transition_probability(
+            source=source, target=target, fitness_function=fitness_function
+        )
+        == 0
+    )
+    source = np.array((1, 1, 0))
+    target = np.array((1, 1, 0))
+    assert (
+        main.compute_transition_probability(
+            source=source, target=target, fitness_function=fitness_function
+        )
+        is None
+    )
+
+
+def test_compute_transition_probability_for_ordered_fitness_function():
+    """"""
+
+    def ordered_fitness_function(state):
+        fitness = np.array([0 for _ in state])
+        zero_encountered = 0
+        one_encountered = 0
+        for position, value in enumerate(state):
+            if value == 0:
+                zero_encountered += 1
+                fitness[position] = zero_encountered + (position % 2)
+            else:
+                one_encountered += 1
+                fitness[position] = one_encountered + (position % 2)
+        return fitness
+
+    source = np.array((0, 1, 0))
+    target = np.array((1, 1, 0))
+    assert (
+        main.compute_transition_probability(
+            source=source, target=target, fitness_function=ordered_fitness_function
+        )
+        == 2 / 15
+    )
+    source = np.array((0, 1, 1))
+    target = np.array((0, 0, 0))
+    assert (
+        main.compute_transition_probability(
+            source=source, target=target, fitness_function=ordered_fitness_function
+        )
+        == 0
+    )
+    source = np.array((1, 1, 0))
+    target = np.array((1, 1, 0))
+    assert (
+        main.compute_transition_probability(
+            source=source, target=target, fitness_function=ordered_fitness_function
+        )
+        is None
+    )
+
+
+def test_compute_transition_probability_for_symbolic_fitness_function():
+    """"""
+
+    def symbolic_return(input):
+        if input == 1:
+            return sym.symbols("x")
+        return sym.symbols("y")
+
+    def symbolic_fitness_function(state):
+        return np.array([symbolic_return(_) for _ in state])
+
+    source = np.array((0, 1, 0))
+    target = np.array((1, 1, 0))
+    x = sym.symbols("x")
+    y = sym.symbols("y")
```

Comment:
```suggestion
    x = sym.Symbol("x")
    y = sym.Symbol("y")
```

(Remind me to explain why!)